scala-thoughts.md

```diff
@@ -0,0 +1,75 @@
+# Reflections on Learning Scala
+
+Scala has been an interesting language to start to learn. What stood out to me
+the most when I was learning it was that the syntax was super flexible - for
+me that was actually very challenging to handle at first. I think in the end I
+tried to just avoid the cases that seemed ambiguous to me, choosing to write
+
+```
+list1.zip(list2).drop(3)
+```
+
+in place of
+
+```
+list1 zip list2 drop 3
+```
+
+I don't think this flexibility is a bad thing (more thoughts on this later),
+but I would note that it has been a long time since I learned a language with
+even a slightly different syntax structure. Its surprising how much a few small
+decisions can do to complicate that structure.
+
+Regarding the functional parts of Scala, I found using them to be very
+satisfying, mostly because I've done enough C++11 and Java to want that
+expressiveness and written enough Python/Haskell/Rust to be comfortable
+using it.
+
+I could imagine that transitioning to Scala from an only-Java perspective would
+be much more challenging.
+
+# The Design Choices of Scala
+
+One of the interesting design choices that has been made in Scala is to allow
+for `A f B` in place of `A.f(b)`. This may seem like just a surface level
+change, but I imagine that this is one of the changes that makes Scala
+appropriate for writing internal DSL's in. There are also some other
+parsing-related choices that contribute to that goal. Even though I don't
+particularly like this decision right now, I imagine that it will grow on me
+over time.
+
```

Comment:
It does seem strange for a language to be so vague, especially if you are used to coding in Python! I agree with you that this is exactly the flexibility we look for when we design a language. A Sca-lable La-nguage! :-)